lib/common/pool.c

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include "pool.h"
+#include <stddef.h>  /* size_t */
+#include <stdlib.h>  /* malloc, calloc, free */
+
+#ifdef ZSTD_PTHREAD
+
+#include <pthread.h>
+
+/* A job is a function and an opaque argument */
+typedef struct POOL_job_s {
+  POOL_function function;
+  void *opaque;
+} POOL_job;
+
+struct POOL_ctx_s {
+    /* Keep track of the threads */
+    pthread_t *threads;
+    size_t numThreads;
+
+    /* The queue is a circular buffer */
+    POOL_job *queue;
+    size_t queueHead;
+    size_t queueTail;
+    size_t queueSize;
+    /* The mutex protects the queue */
+    pthread_mutex_t queueMutex;
+    /* Condition variable for pushers to wait on when the queue is full */
+    pthread_cond_t queuePushCond;
```

Comment:
For example, `pthread_cond_t` is not supported currently by `threading.h` (outside of posix systems).
Which means, there is a need for one of these solutions : 
- extend `threading.h` to support `pthread_cont_t`
- rewrite `pool.c` to not need `pthread_cont_t`
- write a specific `_WIN32` version of `pool.c`